frontend/src/components/visualizations/Visualization.tsx

```diff
@@ -87,6 +88,9 @@ export const Visualization = (props: previewProps) => {
 														"No visualization data or parameters available. " +
 															"Incomplete visualization configuration."
 													);
+													return (
+											<NoVisualizationCard msg = "No visualization data or parameters available. Incomplete visualization configuration."></NoVisualizationCard>
```

Comment:
The reason why cards showing so many times is because this is a loop that check for any registered visualization configuration/ or check for any matching file content type. Some times there are many applicable match.

I would suggest instead create two states to indicate "noVis" and "fileOverThreshold" flag. Then whenever you run into the situation, just turn on those flags.
Have a use Effect to listen to those flag change. If flag is True display the card accordingly.